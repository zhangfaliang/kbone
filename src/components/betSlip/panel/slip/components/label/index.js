import React, { useState } from 'react';
import classnames from 'classnames';
import { BackHeader } from '@/components/header';
import styles from './index.less';

function Label({
  title,
  isEdit: propsIsEdit,
  desc,
  isLogin,
  onChangeEdit,
  className,
  loginRender,
  noLoginRender,
  onClose,
  betCount,
}) {
  const wrapCls = classnames(className, {
    [styles.wrap]: true,
  });

  const [isEdit, setIsEdit] = useState(propsIsEdit);

  return (
    <div className={wrapCls}>
      <div className={styles.header}>
        <BackHeader
          title="串关投注"
          onClose={onClose}
          rightContent={[isLogin ? loginRender : noLoginRender]}
        />
      </div>
      <div className={styles.action}>
        <div className={styles.a}>
          <div className={styles.a_a}>
            <span>{betCount}</span>项
          </div>
          <div
            className={styles.a_b}
            onClick={() => {
              setIsEdit(!isEdit);
              onChangeEdit(!isEdit);
            }}
          >
            {desc}
          </div>
        </div>
      </div>
    </div>
  );
}

Label.defaultProps = {
  isLogin: true,
  loginRender: 'loginRender',
  noLoginRender: 'noLoginRender',
  onClose: () => {},
};

export default Label;
