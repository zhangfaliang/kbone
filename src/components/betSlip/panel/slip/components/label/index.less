.wrap {
  display: flex;
  flex-direction: column;

  .header {
    height: 88px;
  }

  .action {
    display: flex;
    justify-content: space-between;
    padding: 0 24px;
    height: 88px;
    background: #F4F4F5;
    border-radius: 24px 24px 0 0;

    .a {
      display: flex;
      justify-content: space-between;
      align-items: center;
      flex:1;
      .a_a {
        font-family: PingFangSC-Regular;
        font-size: 28px;
        color: #212934;
        letter-spacing: 0;

        span {
          font-size: 56px;
          color: #3D6BFA;
          margin-right: 10px;
        }
      }

      .a_b {
        font-family: PingFangSC-Regular;
        font-size: 28px;
        color: #7A7F85;
        letter-spacing: 0;
        text-align: right;
        display: flex;
      }
    }
  }
}
// @media screen and (min-width: 992px) {
//   .wrap{
//     .header {
//       >div{
//         width: 750px;
//         transform: translateX(0) !important;
//       }
//     }
//   }
 
// }