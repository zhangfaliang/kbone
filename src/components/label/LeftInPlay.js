import React, { Component } from 'react';
import classnames from 'classnames';
import styles from './index.less';
import IconFont from '../iconFont';
import { DEFAULT_ENCODING } from 'crypto';
//TODO 需要添加fonticon 组件 现在等待fonticon添加
class LeftInPlay extends Component {
  static COMPONENT_NAME = 'LEFTINPLAY';
  // handelClickTitleText = e => {
  //   e.stopPropagation();
  //   const { leftText, clickLeftTitleText, type } = this.props;
  //   clickLeftTitleText && clickLeftTitleText({ type, leftText });
  // };

  render() {
    const { liveText, inPlayTex, prefixCls } = this.props;
    const clsNameStr = `${prefixCls || 'default'}-left-title`;

    return (
      <div className={styles[clsNameStr]}>
        {liveText && <span className={styles.liveText}> {liveText}</span>}
        <span className={styles.inPlayTex}>{inPlayTex}</span>
      </div>
    );
  }
}
LeftInPlay.defaultProps = {
  prefixCls: 'live-inplay',
  liveText: '',
  inPlayTex: '',
};

export default LeftInPlay;
