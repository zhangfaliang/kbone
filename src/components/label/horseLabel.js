import React from 'react';
import styles from './horseLabel.less';
import classnames from 'classnames';
import { getStringInTime } from '@/utils/mistiming';

class CountDown extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      text: this.props.text,
    };
  }

  static getDerivedStateFromProps(props, state) {
    if (props.text !== state.text) {
      return { activeKey: props.text };
    }
    return null;
  }
  render() {
    const { text } = this.state;
    return text;
  }
  componentDidMount() {
    const { matchTime } = this.props;
    if (!matchTime) return;
    this.timer = setInterval(() => {
      const { isShowStringInTime, stringInTime } = getStringInTime(matchTime) || {};
      if (isShowStringInTime && stringInTime) {
        this.setState({
          text: isShowStringInTime && stringInTime ? stringInTime : '',
        });
      } else {
        clearInterval(this.timer);
        this.setState({
          text: '',
        });
      }
    }, 1000);
  }
  componentWillUnmount() {
    clearInterval(this.timer);
  }
}

export default ({ labels = [], prefixCls = 'horse', matchTime = '' }) => {
  return (
    <div className={styles[`${prefixCls}-label`]}>
      {labels.map((label, index) => {
        const { labelText, isCountDown, isBtn } = label;

        const clsStr = classnames({
          [styles.text]: true,
          [styles.button]: isBtn,
        });
        return (
          <span className={clsStr} key={index}>
            {isCountDown ? <CountDown text={labelText} matchTime={matchTime} /> : labelText}
          </span>
        );
      })}
    </div>
  );
};
