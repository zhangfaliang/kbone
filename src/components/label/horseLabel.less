@import '../../theme/index.less';
@defaultWidth: 100%;

.horse-label {
  height: @tab-bar-height;
  width: 7.5rem;
  overflow-x: auto;
  // display: flex;
  // justify-content: flex-start;
  text-align: left;
  white-space:nowrap;
  padding: 23px 0 0 0;
  -webkit-overflow-scrolling: touch;
  border-top:1px solid @tab-bar-border-color;
    .text {
    font-family: Akrobat-Regular;
    font-size: 28px;
    color: @aide-color-secondary;
  }
  &.horse-label::-webkit-scrollbar {
    display: none;
  }
  .button {
    height: 40px;
    padding: 5px 20px;
    box-sizing: border-box;
    background: @tab-bar-border-color;
    border-radius: 20px;
    color: #FFFFFF;
  }

  span {
    margin-left: 20px;
    display: inline-block;
    padding:0;
    white-space:nowrap;
  }

  :last-child {
    padding-right: 20px;
  }

  &::-webkit-scrollbar {
    width: 0 !important
  }

}
.nom{
  margin: 0;
  padding: 0;
}