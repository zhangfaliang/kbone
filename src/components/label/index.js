export { default as LeftTitle } from './LeftTitle.js';
export { default as Label } from './Label.js';
export { default as LabelRight } from './LabelRight.js';
export { default as LabelCenter } from './LabelCenter.js';
export { default as LeftInPlay } from './LeftInPlay';
export { default as HorseLabel } from './horseLabel';
