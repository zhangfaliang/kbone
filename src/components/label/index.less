@import '../../theme/index.less';
@defaultHeigth: 80px;

.default-label {
  height: @defaultHeigth;
  // background: @match-group-bg;
  display: flex;
  width: 100%;
  justify-content: space-between;
  align-items: center;
  font-size: 28px;
  // color: @aide-color-secondary;
  
  // padding-left: @h-p-20;
  padding: 0 24px;
  box-sizing: border-box;
  position: relative;
  // background: #F4F4F5;
 >div{
   &:nth-child(2){
     span {
       color: #BCBEC2;
     }
   }
 }
  &.showTBBorder {
    // border-bottom: 1px solid @tab-bar-border-color;
    // border-top: 1px solid @tab-bar-border-color;
  }
}

.default-left-title {
  display: flex;
  justify-content: flex-start;
  align-items: center;
  flex: 1;

  .gameTypeImg {
    display: flex;
    height: 36px;
    width: 36px;
    resize-mode: contain;
    margin-right: 20px;
  }

  .titleText {
    font-size: 28px;
    // color: @aide-color-secondary;
    width:100%; // 处理订单详情
  }
}

.order-scheme-left-title {
 &:extend(.default-left-title all);
  .titleText {
    font-family: Akrobat-Regular;
    // font-size: 32px;
    // color: @aide-color-secondary;
  }
}

.push-before-left-title {
  &:extend(.default-left-title all);

  .titleText {
    font-size: 28px;
    color: @remark-font-color;
  }

  height: 72px;
  box-sizing:border-box;
  padding-left:@h-p-20;
}

.live-inplay-left-title {
  &:extend(.default-left-title all);
  font-size: 32px;
  font-family: Akrobat-Bold;

  .liveText {
    color: #FFE011;
    margin-right: 10px;

  }

  .inPlayTex {
    color: #fff
  }

}

.default-right-table {
  display: flex;
  justify-content: flex-end;
  align-items: center;
  padding-right: 0;
  box-sizing: border-box;
  // margin-right: 20px;
  .table {
    display: flex;
    justify-content: center;
    align-items: center;
    width: 120px;
    height: 80px;
    font-size: 28px;
    color: #8FB8DF;

  }

  .icon {
    display: flex;
    justify-content: center;
    align-items: center;
    height: @iconHeight;
    width: @iconWidth;

    >span::before {
      // color: @aide-color-secondary;
      color:#BCBEC2; 
    }
  }
}

.index-right-table {
  &:extend(.default-right-table all);

  span:last-child {
    margin-right: 20px;
  }
}

.push-before-right-table {
  &:extend(.default-right-table all);

  .table {
    color: @remark-font-color;
  }

  padding-right:0;

}

.preCompetition-before-right-table {
  &:extend(.default-right-table all);

  .table {
    color: @remark-font-color;
  }

  padding-right:0;

}

.live-inplay-right-table {
  &:extend(.default-right-table all);

  .rightText {
    display: block;
    height: @tab-bar-height;
    text-align: center;
    line-height: @tab-bar-height+1;
    font-size: 28px;
    color: #FFFFFF;
  }

  .icon {
    margin-left: 0px;
  }
}

.push-before-label {
  &:extend(.default-left-title all);
  font-size: 28px;
  color: @remark-font-color;
  height: 72px;
}

.other-match-label {
  &:extend(.default-label all);
  padding-right: 5px;
  font-size: 28px;
  color: @remark-font-color;
  background: #FFFFFF;
  border-radius: 8px;
  margin-top: 20px;
}

.bet-detail-label {
  &:extend(.default-label all);
  font-family: Akrobat-Regular;
  font-size: 32px;
  padding-right: 5px;
  // background: #FFFFFF;
  &.showTBBorder {
    // border-bottom: 1px solid  @bet-border-color;
    // border-top: 1px solid  @bet-border-color;
  }
}

.bet-detail-mt20-label {
  &:extend(.bet-detail-label all);
  margin-top:32px;
  // background: #F5F7FF;
}

.order-detail-label {
  &:extend(.bet-detail-label all);
  height: 72px;
  font-size: 30px;
  padding-right: 20px;
  background: #FFFFFF;
  border: none;
  &.showTBBorder {
    border-bottom: 1px solid  @bet-border-color;
    border-top: 1px solid  @bet-border-color;
  }
}

.default-label-center {
  display: flex;
  justify-content: center;
  align-items: center;
  font-size: 28px;
  color: #FFFFFF;
}

.match-detail-label {
  &:extend(.default-label all);
  padding: 0 13px 0 17px;
  color: #FFFFFF;
}

.match-detail-left-title {
  &:extend(.default-left-title all);

  .titleText {
    font-size: 24px;
    color: #FFFFFF;
  }

  flex: none;
}

.other-match-title-left-title {
  &:extend(.default-left-title all);

  .titleText {
    color: #FFFFFF;
  }
}

.white-left-title {
  &:extend(.default-left-title all);

  .titleText {
    color: #FFFFFF;
  }
}

.other-match-item-left-title {
  &:extend(.default-left-title all);

  .titleText {
    color: @remark-font-color;
  }
}

.bet-detail-left-title {
  &:extend(.default-left-title all);
  .titleText {
    font-family: Akrobat-Regular;
    // font-size: 32px;
    font-size: 28px;
    color: @remark-font-color;
    color: #7A7F85;
    margin: 0;
     text-align: left;
  }
}
.bet-detail-mt20-left-title{
  &:extend(.bet-detail-left-title all);
}

.order-detail-left-title {
  &:extend(.default-left-title all);

  .titleText {
    color: @remark-font-color;
  }
}

.match-detail-right-table {
  &:extend(.default-right-table all);

  .icon {
    >span::before {
      color: #fff;
    }
  }

  .activeIcon {
    >span::before {
      color: #FFE011;
    }
  }
}

.bet-detail-right-table {
  &:extend(.default-right-table all);

  .icon {
    margin: 0;

    >span::before {
      font-size: 44px;
    }
  }

}

.order-detail-right-table {
  &:extend(.default-right-table all);

  .rightText {
    color: @remark-font-color;
  }

}
.F5F7FF{
  background: #F5F7FF;
  border-radius: 24px 24px 0 0;
}
.F4F4F5 {
  background: #F4F4F5;
  border-radius: 24px 24px 0 0;
  font-family: SFProText-Regular;
  font-size: 28px;
  color: #7A7F85 ;
  letter-spacing: 0;
}
