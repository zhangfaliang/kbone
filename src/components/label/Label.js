import React, { Component } from 'react';
import LeftTitle from './LeftTitle';
import LabelRight from './LabelRight';
import LabelCenter from './LabelCenter';
import LeftInPlay from './LeftInPlay';
import ResultImg from '../order/orderCard/ResultImg';
import styles from './index.less';
import classnames from 'classnames';
class Label extends Component {
  static COMPONENT_NAME = 'LABEL';
  handelClickTitleText = e => {
    e.preventDefault();

    const { clickTitleText, labelType, disabled } = this.props;

    this.props.clickTitleText && !disabled && clickTitleText(labelType);
  };
  render() {
    const {
      children,
      prefixCls,
      leftNode,
      rightNode,
      showTBBorder,
      clickTitleText,
      bkgdColorCls,
    } = this.props;
    const childrens = React.Children.map(children, option => {
      const { ...other } = option.props;
      if (option.type.COMPONENT_NAME === 'LEFTNODE') {
        return <LeftTitle {...other} clickLeftTitleText={this.props.clickLeftTitleText} />;
      } else if (option.type.COMPONENT_NAME === 'LABLEDCENTER') {
        return <LabelCenter {...other} />;
      } else if (option.type.COMPONENT_NAME === 'LABELRIGHT') {
        return <LabelRight {...other} handleRightIcon={this.props.handleRightIcon} />;
      } else if (option.type.COMPONENT_NAME === 'LEFTINPLAY') {
        return <LeftInPlay {...other} />;
      } else if (option.type.COMPONENT_NAME === 'RESULTIMG') {
        return <ResultImg {...other} />;
      }
    });

    const clsNameStr = `${prefixCls || 'default'}-label`;
    const claName = classnames({
      [styles[clsNameStr]]: true,
      [styles['showTBBorder']]: showTBBorder,
      [styles[`${bkgdColorCls}`]]: bkgdColorCls ? true : false,
    });
    return (
      <div onClick={this.handelClickTitleText} className={claName}>
        {childrens}
      </div>
    );
  }
}
Label.defaultProps = {
  prefixCls: 'default',
  clickLeftTitleText: params => {
    console.log('clickLeftTitleText');
    return 'clickLeftTitleText';
  },
  showTBBorder: false,
  clickTitleText: params => {
    console.log('clickTitleText');
    return 'clickTitleText';
  },
  handleRightIcon: () => {
    console.log('handleRightIcon');
    return 'handleRightIcon';
  },
  labelType: '电竞',
  disabled: false,
};
export default Label;
