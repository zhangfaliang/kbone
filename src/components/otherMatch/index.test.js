import toJson from 'enzyme-to-json';
import { render, mount } from 'enzyme';
import OtherMatch from './index';
import { Label } from '../label';
describe('OtherMatch', async () => {
  it('base', () => {
    const warpper = render(<OtherMatch />);
    expect(toJson(warpper)).toMatchSnapshot();
  });
  it('basePrefixNull', () => {
    const warpper = render(<OtherMatch prefixCls={null} />);
    expect(toJson(warpper)).toMatchSnapshot();
  });
  it('clickLeftTitleText', () => {
    expect.assertions(2);
    const warpper = mount(<OtherMatch />);
    const clickLeftTitleText = warpper.prop('clickLeftTitleText');
    const clickTitleText = warpper.prop('clickTitleText');
    const clickLeftTitleTextFn = jest.fn(clickLeftTitleText);
    expect(clickLeftTitleTextFn()).toBe('clickLeftTitleText');
    const clickTitleTextFn = jest.fn(clickTitleText);
    expect(clickTitleTextFn()).toBe('clickTitleText');
  });

  it('clickLeftTitleText1', () => {
    expect.assertions(1);
    const warpper = render(
      <OtherMatch>
        <Label />
        <Label />
      </OtherMatch>
    );

    expect(toJson(warpper)).toMatchSnapshot();
  });
});
