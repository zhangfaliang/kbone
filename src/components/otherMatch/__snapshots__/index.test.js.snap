// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OtherMatch base 1`] = `
<div
  class="default-other-match"
/>
`;

exports[`OtherMatch basePrefixNull 1`] = `
<div
  class="default-other-match"
/>
`;

exports[`OtherMatch clickLeftTitleText1 1`] = `
<div
  class="default-other-match"
>
  <div
    class="default-label"
  />
  <div
    class="default-label"
  />
</div>
`;
